### PR TITLE
Support ForEachVariableStatement in stackalloc analyzer

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDoNotUseStackallocInLoops.cs [hunk 1]
```diff
@@ -71,7 +72,7 @@ static bool ShouldWarn(IOperation? op, SyntaxNode node)
                                         foreach (IOperation child in block.Operations)
                                         {
                                             if (child.Syntax.SpanStart > node.SpanStart &&
-                                                (child is IReturnOperation || (child is IBranchOperation branch && branch.BranchKind == BranchKind.Break)))
+                                                (child is IReturnOperation or IBranchOperation { BranchKind: BranchKind.Break }))
```
Comment (Youssef1313) [inline src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDoNotUseStackallocInLoops.cs:75]: No behavior change here. only a minor stylish change.

File: src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDoNotUseStackallocInLoops.cs [hunk 2]
```diff
@@ -88,7 +89,7 @@ static bool ShouldWarn(IOperation? op, SyntaxNode node)
                             }
 
                             // Warn as needed.
-                            if (ShouldWarn(ctx.SemanticModel.GetOperationWalkingUpParentChain(ctx.Node, default), ctx.Node))
+                            if (ShouldWarn(ctx.SemanticModel.GetOperationWalkingUpParentChain(ctx.Node, ctx.CancellationToken), ctx.Node))
```
Comment (Youssef1313) [inline src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDoNotUseStackallocInLoops.cs:92]: This is an unrelated improvement.

File: src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDoNotUseStackallocInLoops.cs [hunk 3]
```diff
@@ -38,7 +38,8 @@ public sealed override void Initialize(AnalysisContext context)
                         // foreach loops are special, in that as with other loops we don't want stackallocs
                         // in the body of the loop, but in the expression of a foreach is ok.
                         case SyntaxKind.ForEachStatement:
-                            ForEachStatementSyntax foreachStatement = (ForEachStatementSyntax)node;
+                        case SyntaxKind.ForEachVariableStatement:
+                            var foreachStatement = (CommonForEachStatementSyntax)node;
```
Comment (Youssef1313) [inline src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDoNotUseStackallocInLoops.cs:42]: This is the main fix.

**Review by mavasani — APPROVED** [review]

**Review by stephentoub — COMMENTED** [review]
Thanks!
